cas-server-support-x509/src/main/java/org/jasig/cas/adaptors/x509/authentication/principal/X509SubjectAlternativeNameUPNPrincipalResolver.java

```diff
@@ -89,11 +91,23 @@ protected String resolvePrincipalInternal(final X509Certificate certificate) {
     private String getUPNStringFromSequence(final ASN1Sequence seq) {
         if (seq != null) {
             // First in sequence is the object identifier, that we must check
-            final DERObjectIdentifier id = DERObjectIdentifier.getInstance(seq.getObjectAt(0));
+            final ASN1ObjectIdentifier id = ASN1ObjectIdentifier.getInstance(seq.getObjectAt(0));
             if (id != null && UPN_OBJECTID.equals(id.getId())) {
                 final ASN1TaggedObject obj = (ASN1TaggedObject) seq.getObjectAt(1);
-                final DERUTF8String str = DERUTF8String.getInstance(obj.getObject());
-                return str.getString();
+                ASN1Primitive prim = obj.getObject();
+                
+                // Due to bug in java cert.getSubjectAltName, it can be tagged an extra time
+                if (prim instanceof ASN1TaggedObject) {
+                    prim = ASN1TaggedObject.getInstance(((ASN1TaggedObject) prim)).getObject();
+                }
+
+                if (prim instanceof ASN1OctetString) {
```

Comment:
Nope, you can't do that. You need to double check that if it is encapsulated as an ASN1 tagged object, and get the inner object to see what type of string it is. You need that due to a java bug, sometimes it is tagged twice, so (ASN1TaggedObject) seq.getObjectAt(1) and that first if (ASN1TaggedObject.getInstance(((ASN1TaggedObject) prim)).getObject();) does that.